blog-posts/emails/article.md

```diff
@@ -0,0 +1,331 @@
+
+I modified my email's header MJML code to include an image:
+
+```xml
+<mj-section background-color="#F48668" border-radius="8px 8px 0 0" padding="0">
+  <mj-column>
+    <mj-image width="20px" src="cid:my-logo@pchol.fr" alt="My logo"/>
+  </mj-column>
+</mj-section>
+```
+
+Then, in the lambda I use to send my emails, I can use the Nodemailer library to translate my MJML template into a raw email, and to add attachments to my message.
+
+```js
+import * as aws from '@aws-sdk/client-ses';
+import nodemailer from 'nodemailer';
+import myHtmlTemplate from './myHtmlTemplate'; // imported as a string
+
+export const handler = async () => {
+  const sesClient = new aws.SESClient({});
+  const transporter = nodemailer.createTransport({
+    SES: { ses: sesClient, aws },
+  });
+
+  await transporter.sendEmail({
+    from: 'notifications@pchol.fr', // my domain name
+    to: '<recipient>',
+    subject: 'This a raw email test',
+    text: 'This a raw email test',
+    html: myHtmlTemplate.replace('{{username}}', 'Pchol').replace('{{password}}', 'very_secret_password'), // manually replace parameters
+    attachments: [
+      {
+        filename: 'pchol-logo.png',
+        path: '/opt/pchol-logo.png', // image stored in a Lambda layer
+        cid: 'my-logo@pchol.fr', // same as in the template
+      },
+    ],
+  });
+
+  return 'OK!';
+};
+```
+
+The files that are attached to my mails come from a Lambda layer. These layers can be easily set on the different available frameworks.
+
+Using additional parameters like `list` or `headers`, you can specify custom headers for your emails (which is impossible using templated emails). For instance, to add a `List-unsubscribe` header (the last obstacle to our mail-tester perfect score), you can add the following code to our handler:
+
+```typescript
+list: {
+  unsubscribe: {
+    url: 'http://www.pchol.fr',
+    comment: 'Unsubscribe from this notifications',
+  },
+},
+```
+
+Here is the final result, containing the logo as an attachment and the `List-unsubscribe` header. The full mail, with html styles, attachment and custom headers obtains a 100% score on mail-tester when I send it from my custom domain based SES identity.
+
+![Final email with attachment](./assets/email-final.png 'Final email with attachment')
+
+## Conclusion
+
+From handling AWS sender reputation, to defining a responsive UI, going trough setting appropriate DNS records on our domain name, I tried to introduce you to three common issues and their solutions. This problems are often encountered while sending naive emails with SES, and make a difference between professional looking messaging and spams.
+
+There are many other things that can be improved, like dark mode management, but I'll let you dig now that you have been introduced to these concepts!
```

Comment:
I wouldn't choose this concluding sentence :sweat_smile: Maybe add a link to "dark mode management". Otherwise it feels like you are too lazy to talk about it.